Library/Homebrew/shims/super/cc

```diff
@@ -308,7 +308,10 @@ class Cmd
 
     args << "-w" unless configure?
     args << "-#{ENV["HOMEBREW_OPTIMIZATION_LEVEL"]}"
-    args.concat(optflags) unless runtime_cpu_detection?
+    optflags.each do |optflag|
+      flag = optflag.split("=").first
+      args << optflag if !runtime_cpu_detection? || @args.none? { |arg| arg.start_with?(flag) }
```

Comment:
@carlocab could this get a Ruby comment in a follow-up PR? I don't really understand what this is doing or why it's necessary 😅

Reply:
#19555